aplijobs/birdwoot
# Feature | Add sentry config and implementation

# Pull Request Template

## Description

This PR adds basic Sentry reports on main API requests for chat and dashboard
- Adds env vars
- Add exception file for `Sentry capture exception`
- Modify Sentry init in Dashboard App

## Type of change

Please delete options that are not relevant.


## How Has This Been Tested?

To test that errors are being reported to Sentry you need to add the Client Side Keys for each project. 
Remember that for development we use the projects with the prefix `_dev`

CONVERSATIONS_DEV_BIRDWOOT_CHAT
CONVERSATIONS_DEV_BIRDWOOT_DASHBOARD

- Modify env vars values: 
	#VUE_APP_SENTRY_DSN_CHAT= CONVERSATIONS_DEV_BIRDWOOT_CHAT (Client dsn key)
	#VUE_APP_SENTRY_DSN_DASHBOARD=CONVERSATIONS_DEV_BIRDWOOT_DASHBOARD (Client dsn key)

- Go to this file `app/javascript/widget/store/modules/conversation/actions.js`
- And throw an error in `sendMessageWithData` function (or any other)
f.e 
```javascript
  sendMessageWithData: async ({ commit }, message) => {
    const { id, content, meta = {} } = message;

    commit('pushMessageToConversation', message);
    commit('updateMessageMeta', { id, meta: { ...meta, error: '' } });
    try {
      throw new Error('TEST ERROR');
      const { data } = await sendMessageAPI(content);
      commit('deleteMessage', message.id);
      commit('pushMessageToConversation', { ...data, status: 'sent' });
    } catch (error) {
      captureSentryException(error);
      commit('pushMessageToConversation', { ...message, status: 'failed' });
      commit('updateMessageMeta', {
        id,
        meta: { ...meta, error: '' },
      });
    }
  },
```
- Uncomment line 15 in this file `app/javascript/packs/widget.js`

- Run birdwoot
- Try to write a message and send it, the error must be applied
- Go to Sentry project and verify error is being reported

## Checklist:
